client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -200,8 +200,11 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
-        Map<SegmentWithRange, Long> segments = getSegmentsForStreams(controller, config);
-        synchronizer.updateStateUnconditionally(new ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
+        synchronizer.fetchUpdates();
```

Comment:
as suggested above, please add `updatingConfig` flag to RGState and change resetReaderGroup to:
```
resetRG(RGConfig newConfig) {
ss.updateState(set updatingConfig=true)
controller.updateRG()
ss.updateStateUnconditionally(RGInit... updatingConfig=false)
}
```